lua/telescope/previewers.lua

```diff
@@ -288,52 +288,19 @@ previewers.vim_buffer = defaulter(function(_)
     end,
 
     preview_fn = function(self, entry, status)
-      -- TODO: Consider using path here? Might not work otherwise.
-      local filename = entry.filename
-
-      if filename == nil then
-        filename = entry.path
-      end
-
-      if filename == nil then
-        local value = entry.value
-        filename = vim.split(value, ":")[1]
-      end
-
-      if filename == nil then
-        return
-      end
+      local bufnr = tonumber(entry.bufnr)
 
-      log.info("Previewing File:", filename)
-
-      local bufnr = vim.fn.bufnr(filename)
-      if bufnr == -1 then
-        -- TODO: Is this the best way to load the buffer?... I'm not sure tbh
-        bufnr = vim.fn.bufadd(filename)
+      if not vim.api.nvim_buf_is_loaded(bufnr) then
         vim.fn.bufload(bufnr)
-
-        vim.cmd([[doautocmd filetypedetect BufRead ]] .. filename)
```

Comment:
does calling bufload do this?

Reply:
yes i think so. I have highlighting for unloaded buffers and it seems like for loaded buffers i don't have to do anything